airflow/jobs/scheduler_job_runner.py

```diff
@@ -1649,11 +1785,20 @@ def _fail_tasks_stuck_in_queued(self, session: Session = NEW_SESSION) -> None:
                 cleaned_up_task_instances = set(executor.cleanup_stuck_queued_tasks(tis=stuck_tis))
                 for ti in stuck_tis:
                     if repr(ti) in cleaned_up_task_instances:
-                        self._task_context_logger.warning(
+                        self.log.warning(
                             "Marking task instance %s stuck in queued as failed. "
                             "If the task instance has available retries, it will be retried.",
                             ti,
-                            ti=ti,
+                        )
+                        session.add(
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/rohankokkulabito/airflow/pull/6#issuecomment-2251038846>#720bfc</a><span> - 07/25/2024, 05:35 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The addition of the log entry to the database within the loop can lead to performance degradation due to multiple database writes. This can be optimized by batching the log entries and performing a single database write.
</div>
<div id="fix">
<b>Fix:</b> Collect all log entries in a list and add them to the session in a single batch operation after the loop.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -1792,11 +1792,12 @@
 +                        log_entries = []
 +                        log_entries.append(
 +                            Log(
 +                                event="stuck in queued",
 +                                task_instance=ti.key,
 +                                extra=(
 +                                    "Task will be marked as failed. If the task instance has "
 +                                    "available retries, it will be retried."
 +                                ),
 +                            )
 +                        )
 +                    session.add_all(log_entries)
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged